### PR TITLE
Updates addon to adhere to RFC 176

### PR DESCRIPTION
**Pull request by locks**
This PR attempts to migrate all of the documentation and imports to adhere to [RFC #176](https://github.com/emberjs/rfcs/blob/master/text/0176-javascript-module-api.md).

- [x] Updates documentation
- [x] Updates code base
- [x] Add modules polyfill to the globals/bower build

### CODE REVIEW

**Comment by rwjblue** [comment]
@locks - I believe we need to add [babel-plugin-ember-modules-api-polyfill](https://github.com/ember-cli/babel-plugin-ember-modules-api-polyfill) to [the globals/bower build](https://github.com/emberjs/data/blob/d35feb17fd67f4ff2016f86b45d9e81b0f53a5d4/lib/babel-build.js#L51-L64). Otherwise these changes will only work for consumers using ember-data as an addon.

**Comment by stefanpenner** [comment]
@locks nice. I believe the bower build stuff will also need to be update (just chatting with @rwjblue about this)

**Comment by tylerturdenpants** [comment]
Some conversation was happening in slack in regards to $.parseJSON.  Could this PR possibly adress moving to the native parse?  Or should that be another PR?

**Comment by locks** [comment]
@tylerturdenpants this PR already touches 161 files, I think it would be best to keep it focused on imports/exports.

**Comment by stefanpenner** [comment]
@locks any plans to finish this?

**Comment by locks** [comment]
@stefanpenner yessir! I'll try to finish before the weekend, but next week is more likely.

**Comment by locks** [comment]
@stefanpenner @rwjblue please review 😁

**Comment by rwjblue** [comment]
@locks - Sorry for the delays, I believe we are unblocked here now that https://github.com/emberjs/data/pull/5123 has landed.  Would you mind rebasing?

**Comment by locks** [comment]
Kicked CI and now it's green, take that as you will.

File: MODULE_REPORT.md [hunk 1]
```diff
@@ -0,0 +1,1372 @@
+## Module Report
```
[the remark this answers is not in the document]
Comment (locks) [reply MODULE_REPORT.md]: Confirm, it was my intention for `MODULE_REPORT.md` to document what work might still need to be done.

**Comment by bmac** [comment]
@locks would you mind rebasing this pr?

**Comment by locks** [comment]
Rebased and updated documentation code blocks as well.

File: tests/integration/adapter/store-adapter-test.js [hunk 2]
```diff
@@ -1,25 +1,18 @@
...
- and its adapter.
-
- Typically, when a method is invoked on the store, it calls a related
- method on its adapter. The adapter notifies the store that it has
- completed the assigned task, either synchronously or asynchronously,
- by calling a method on the store.
-
- These tests ensure that the proper methods get called, and, if applicable,
- the given record or record array changes state appropriately.
-*/
```
Comment (bmac) [inline tests/integration/adapter/store-adapter-test.js]: Why did this comment get removed?

**Pull request merged**